프로젝트 빌드(./gradlew build) 실패

## ✍ Description
gradle을 이용하여 프로젝트 빌드 `./gradlew build`을  시도할 경우 작성한 테스트코드의 실패 메시지와 함께 빌드가 실패됨.
테스트코드를 돌려보면서 확인결과 이상없음에도 불구하고 계속 같은 현상 확인.


## 재현 과정
어떤 상황에 버그️가 발생하는지 작성하세요.
1. 터미널에 `./gradlew build` 입력시 다음과 같은 메시지를 받을 수 있다.
```
34 tests completed, 7 failed

> Task :test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///Users/jaehyunkim/Dev/Hanbang-E/build/reports/tests/test/index.html

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org 
```
## 📸 스크린샷
![image](https://user-images.githubusercontent.com/28504937/212023958-13f6ed94-419c-4af1-9dfe-423f32365323.png)


## 예상하는 정상 작동
모든 테스트가 통과되고 빌드 성공

--- a/src/test/java/com/hanbang/e/order/controller/OrderControllerTest.java
+++ b/src/test/java/com/hanbang/e/order/controller/OrderControllerTest.java
@@ -3,7 +3,6 @@ package com.hanbang.e.order.controller;
 import static org.assertj.core.api.Assertions.*;
 
 import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
@@ -62,7 +61,7 @@ class OrderControllerTest {
 	@Test
 	void doOrderTest() throws JsonProcessingException {
 		/* given - 데이터 준비 */
-		headers.add("Authorization", jwtUtil.createToken(1L));
+
 		Member dummyMember = new Member("tj087@naver.com", "1234", "제주시");
 		Product product1 = Product.builder()
 			.productName("아이폰11")
@@ -73,6 +72,7 @@ class OrderControllerTest {
 			.onSale(true)
 			.build();
 		memberRepository.save(dummyMember);
+		headers.add("Authorization", jwtUtil.createToken(dummyMember.getMemberId()));
 		productRepository.save(product1);
 
 		OrderReq order = new OrderReq(1);
@@ -80,7 +80,7 @@ class OrderControllerTest {
 
 		/* when - 테스트 실행 */
 		HttpEntity<String> request = new HttpEntity<>(body, headers);
-		ResponseEntity<String> response = rt.exchange("/api/order?productId=1", HttpMethod.POST, request, String.class);
+		ResponseEntity<String> response = rt.exchange("/api/order?productId="+product1.getProductId(), HttpMethod.POST, request, String.class);
 
 		/* then - 검증 */
 		DocumentContext dc = JsonPath.parse(response.getBody());
@@ -96,7 +96,6 @@ class OrderControllerTest {
 	@Test
 	void getMyOrderListTest() throws JsonProcessingException {
 		/* given - 데이터 준비 */
-		headers.add("Authorization", jwtUtil.createToken(1L));
 		Product product1 = new Product("화장품", 23000L, "http://화장품.jpg", 50, 50, true);
 		productRepository.save(product1);
 
@@ -108,6 +107,7 @@ class OrderControllerTest {
 
 		Member member = new Member("mina@naver.com", "12345", "제주도");
 		memberRepository.save(member);
+		headers.add("Authorization", jwtUtil.createToken(member.getMemberId()));
 
 		Orders order1 = Orders.builder()
 			.destination(member.getAddress())
@@ -149,7 +149,7 @@ class OrderControllerTest {
 		Integer quantity = dc.read("$.data[1].quantity");
 		Integer productPrice = dc.read("$.data[1].productPrice");
 		Integer totalPrice = dc.read("$.data[1].totalPrice");
-		Integer productId = dc.read("$.data[1].productId");
+		Long productId = dc.read("$.data[1].productId", Long.class);
 		String productName = dc.read("$.data[1].productName");
 		String img = dc.read("$.data[1].img");
 
@@ -160,7 +160,7 @@ class OrderControllerTest {
 		assertThat(quantity).isEqualTo(order2.getQuantity());
 		assertThat(productPrice).isEqualTo(36000);
 		assertThat(totalPrice).isEqualTo(72000);
-		assertThat(productId).isEqualTo(2);
+		assertThat(productId).isEqualTo(order2.getProduct().getProductId());
 		assertThat(productName).isEqualTo(order2.getProduct().getProductName());
 		assertThat(img).isEqualTo(order2.getProduct().getImg());
 
@@ -170,7 +170,6 @@ class OrderControllerTest {
 	@Test
 	public void deleteOrderTest() throws JsonProcessingException {
 		/* given - 데이터 준비 */
-		headers.add("Authorization", jwtUtil.createToken(1L));
 		Member dummyMember = new Member("tj087@naver.com", "1234", "제주시");
 		Product product1 = Product.builder()
 			.productName("아이폰11")
@@ -181,6 +180,7 @@ class OrderControllerTest {
 			.onSale(true)
 			.build();
 		memberRepository.save(dummyMember);
+		headers.add("Authorization", jwtUtil.createToken(dummyMember.getMemberId()));
 		productRepository.save(product1);
 		Orders dummyOrder = Orders.builder()
 			.destination(dummyMember.getAddress())

--- a/src/test/java/com/hanbang/e/product/controller/ProductControllerTest.java
+++ b/src/test/java/com/hanbang/e/product/controller/ProductControllerTest.java
@@ -41,8 +41,42 @@ class ProductControllerTest {
 		headers.setContentType(MediaType.APPLICATION_JSON);
 	}
 
-	@BeforeEach
-	public void dataSet() {
+	@AfterEach
+	public void dataClear() {
+		productRepository.deleteAll();
+	}
+
+	@DisplayName("상품 상세조회")
+	@Test
+	public void getProductDetailsTest() {
+		/* given - 데이터 준비 */
+		Product product1 = Product.builder()
+			.productName("아이폰11")
+			.price(500000L)
+			.img("http....")
+			.stock(10)
+			.sales(50)
+			.onSale(true)
+			.build();
+		productRepository.save(product1);
+
+		/* when - 테스트 실행 */
+		HttpEntity<String> request = new HttpEntity<>(null, headers);
+		ResponseEntity<String> response = rt.exchange("/api/product/details/"+ product1.getProductId(), HttpMethod.GET, request, String.class);
+
+		/* then - 검증 */
+		DocumentContext dc = JsonPath.parse(response.getBody());
+		String result = dc.read("$.result");
+		String name = dc.read("$.data.name");
+
+		assertThat(result).isEqualTo("success");
+		assertThat(name).isEqualTo("아이폰11");
+	}
+
+	@DisplayName("상품 검색하기, 낮은 가격순 조회")
+	@Test
+	public void searchProductOrderByLowToHighTest() {
+		/* given - 데이터 준비 */
 		Product product1 = Product.builder()
 			.productName("아이폰11")
 			.price(500000L)
@@ -67,39 +101,10 @@ class ProductControllerTest {
 			.sales(75)
 			.onSale(true)
 			.build();
+
 		productRepository.save(product1);
 		productRepository.save(product2);
 		productRepository.save(product3);
-	}
-
-	@AfterEach
-	public void dataClear() {
-		productRepository.deleteAll();
-	}
-
-	@DisplayName("상품 상세조회")
-	@Test
-	public void getProductDetailsTest() {
-		/* given - 데이터 준비 */
-
-		/* when - 테스트 실행 */
-		HttpEntity<String> request = new HttpEntity<>(null, headers);
-		ResponseEntity<String> response = rt.exchange("/api/product/details/1", HttpMethod.GET, request, String.class);
-
-		System.out.println(response);
-		/* then - 검증 */
-		DocumentContext dc = JsonPath.parse(response.getBody());
-		String result = dc.read("$.result");
-		String name = dc.read("$.data.name");
-
-		assertThat(result).isEqualTo("success");
-		assertThat(name).isEqualTo("아이폰11");
-	}
-
-	@DisplayName("상품 검색하기, 낮은 가격순 조회")
-	@Test
-	public void searchProductOrderByLowToHighTest() {
-		/* given - 데이터 준비 */
 
 		/* when - 테스트 실행 */
 		HttpEntity<String> request = new HttpEntity<>(null, headers);
@@ -124,6 +129,34 @@ class ProductControllerTest {
 	@Test
 	public void searchProductOrderByHighToLowTest() {
 		/* given - 데이터 준비 */
+		Product product1 = Product.builder()
+			.productName("아이폰11")
+			.price(500000L)
+			.img("http....")
+			.stock(10)
+			.sales(50)
+			.onSale(true)
+			.build();
+		Product product2 = Product.builder()
+			.productName("아이폰12")
+			.price(1000000L)
+			.img("http....")
+			.stock(15)
+			.sales(100)
+			.onSale(true)
+			.build();
+		Product product3 = Product.builder()
+			.productName("아이폰13")
+			.price(1200000L)
+			.img("http....")
+			.stock(10)
+			.sales(75)
+			.onSale(true)
+			.build();
+
+		productRepository.save(product1);
+		productRepository.save(product2);
+		productRepository.save(product3);
 
 		/* when - 테스트 실행 */
 		HttpEntity<String> request = new HttpEntity<>(null, headers);
@@ -148,6 +181,34 @@ class ProductControllerTest {
 	@Test
 	public void searchProductOrderByBestSellingTest() {
 		/* given - 데이터 준비 */
+		Product product1 = Product.builder()
+			.productName("아이폰11")
+			.price(500000L)
+			.img("http....")
+			.stock(10)
+			.sales(50)
+			.onSale(true)
+			.build();
+		Product product2 = Product.builder()
+			.productName("아이폰12")
+			.price(1000000L)
+			.img("http....")
+			.stock(15)
+			.sales(100)
+			.onSale(true)
+			.build();
+		Product product3 = Product.builder()
+			.productName("아이폰13")
+			.price(1200000L)
+			.img("http....")
+			.stock(10)
+			.sales(75)
+			.onSale(true)
+			.build();
+
+		productRepository.save(product1);
+		productRepository.save(product2);
+		productRepository.save(product3);
 
 		/* when - 테스트 실행 */
 		HttpEntity<String> request = new HttpEntity<>(null, headers);

